src/RequestHandler/outofprocess/forwardinghandler.cpp

```diff
@@ -1492,76 +1575,94 @@ None
     }
 
 Finished:
-
-    if (fLockAcquired)
+    //
+    // Since we use TLS to guard WinHttp operation, call PostCompletion instead of
+    // IndicateCompletion to allow cleaning up the TLS before thread reuse.
+    // Never post after the request has been finished for whatever reason
+    //
+    // Only postCompletion after all WinHttp handle got closed,
+    // i.e., received WINHTTP_CALLBACK_STATUS_HANDLE_CLOSING callback for all handles
```

Comment:
Consider to clarify "all handles": essentially it means two winhttp handles: http and websocket

Reply:
done